internal/servers/controller/controller.go

```diff
@@ -151,6 +164,52 @@ func New(conf *Config) (*Controller, error) {
 	return c, nil
 }
 
+// setStatusGracePeriod returns the status grace period setting for this
+// controller, in seconds.
+//
+// The grace period is the length of time we allow connections to run
+// on a worker in the event of an error sending status updates. The
+// period is defined the length of time since the last successful
+// update.
+//
+// The setting is derived from one of the following:
+//
+//   * Through configuration,
+//   * BOUNDARY_STATUS_GRACE_PERIOD, if defined, can be set to an
+//   integer value to define the setting.
+//   * If either of these is missing, the default (15 seconds) is
+//   used.
+//
+// The minimum setting for this value is the default setting. Values
+// below this will be reset to the default.
+func (c *Controller) setStatusGracePeriod() {
```

Comment:
Rather than copy this code between controllers and workers, please make it a function of `base.Server` that checks the env var and uses that or uses the default (similar to how we set up listeners, controllers, etc.), and sets it in the Server struct. Since both worker and controller have that in their local `Config` object they can share the logic.